controllers/pattern_controller.go

```diff
@@ -190,6 +194,19 @@ func (r *PatternReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ct
 		// Someone manually removed the owner ref
 		return r.actionPerformed(qualifiedInstance, "create application", fmt.Errorf("We no longer own Application %q", targetApp.Name))
 	}
+	// Unseal the vault
+	if qualifiedInstance.Spec.InsecureVaultUnseal {
+		log.Printf("Insecure Vault set to true")
+		if err := vaultInitialize(r.config, r.fullClient); err != nil {
+			return r.actionPerformed(qualifiedInstance, "Initialize vault not successful", err)
+		}
```

Comment:
```suggestion
		if err, ok := vaultInitialize(r.config, r.fullClient); err != nil || ok {
			return r.actionPerformed(qualifiedInstance, "Initialize vault", err)
		}
```

Reply:
yes to changing the signature.
or going with the above as long as its in an if-clause that checks if initialization is needed

Reply:
Ok, I think I addressed this